client/oidc.go

```diff
@@ -76,7 +92,54 @@ type OpenIdProvider interface {
 	VerifyNonGQSig(ctx context.Context, idt []byte, expectedNonce string) error
 }
 
+// Interface for interacting with the OP (OpenID Provider)
+type BrowserOpenIdProvider interface {
+	OpenIdProvider
+	HookHTTPSession(h http.HandlerFunc)
+}
```

Comment:
I don't think this interface belongs in this file?

Reply:
What do you suggest we move it to?

Reply:
Typically it goes closest to where it's used so likely `client.go`. I just didn't know if you had put it in this file for some reason that I missed.

Reply:
I put it in oidc.go since it is an OIDC provider and it makes the most sense to me to put is directly below the `OpenIDProvider` interface definition which it extends. That said, it also makes sense to put in it `client.go`, I had it there at point. I have no objection to moving it to `client.go` if you want me to move it.